src/Symfony/Component/String/AbstractString.php

```diff
@@ -0,0 +1,555 @@
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function trimLeft(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): self
+    {
+        if (!preg_match('//u', $chars)) {
+            throw new InvalidArgumentException('Invalid UTF-8 chars.');
+        }
+        $chars = preg_quote($chars);
+
+        $str = clone $this;
+        $str->string = preg_replace("{^[$chars]++}u", '', $str->string);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function trimRight(string $chars = " \t\n\r\0\x0B\x0C\u{A0}\u{FEFF}"): self
+    {
+        if (!preg_match('//u', $chars)) {
+            throw new InvalidArgumentException('Invalid UTF-8 chars.');
+        }
+        $chars = preg_quote($chars);
+
+        $str = clone $this;
+        $str->string = preg_replace("{[$chars]++$}u", '', $str->string);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function truncate(int $length, string $ellipses = ''): self
+    {
+        $stringLength = $this->length();
+
+        if ($stringLength <= $length) {
+            return $this;
+        }
+
+        $ellipsesLength = '' !== $ellipses ? (new static($ellipses, $this->locale))->length() : 0;
+
+        if ($length < $ellipsesLength) {
+            $ellipsesLength = 0;
+        }
+
+        $str = $this->substr(0, $length - $ellipsesLength);
+
+        return $ellipsesLength ? $str->trimRight()->append($ellipses) : $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function wordwrap(int $width = 75, string $break = "\n", bool $cut = false): self
+    {
+        $lines = $this->split($break);
+        $chars = [];
+        $mask = '';
+
+        if (1 === \count($lines) && '' === $lines[0]->string) {
+            return $lines[0];
+        }
+
+        foreach ($lines as $i => $line) {
+            if ($i) {
+                $chars[] = $break;
+                $mask .= '#';
+            }
+
+            foreach ($line->chunk() as $char) {
+                $chars[] = $char->string;
+                $mask .= ' ' === $char->string ? ' ' : '?';
+            }
+        }
+
+        $string = '';
+        $j = 0;
+        $b = $i = -1;
+        $mask = wordwrap($mask, $width, '#', $cut);
+
+        while (false !== $b = strpos($mask, '#', $b + 1)) {
+            for (++$i; $i < $b; ++$i) {
+                $string .= $chars[$j];
+                unset($chars[$j++]);
+            }
+
+            if ($break === $chars[$j] || ' ' === $chars[$j]) {
+                unset($chars[$j++]);
+            }
+
+            $string .= $break;
+        }
+
+        $str = clone $this;
+        $str->string = $string.implode('', $chars);
+
+        return $str;
+    }
+
+    /**
+     * @return static
+     */
+    public function withLocale(string $locale): self
+    {
+        $str = clone $this;
+        $str->locale = str_replace('-', '_', strtolower($locale));
```

Comment:
im not sure where this is used, but is the strtolower intended given e.g. `en_US` being a locale?

Reply:
The goal of using `strtolower()` is for internal normalization purpose. I might remove this actually, it's not used currently.